### PR TITLE
chore: add university type and minor fixes

### LINKED ISSUE #15: REQUEST: University type
As a user, I want to browser to:

- [ ] be able to browse all federal universities only
- [ ] be able to browse all state universities only
- [ ] be able to browse all private universities only

I think adding a key-value pair `type: 'federal'` etc., will solve that.

### PR DESCRIPTION
**Pull request by stephengade**
In this commit:

1. I moved all types `types/university.type.ts`: as the application grows bigger, having the types in the API file may become too annoying to manage.
2. I added `UniversityType` type - (union ==> `"federal" | "state" | "private")`
3. I updated all universities in the existing university-json file
4. I added `ekiti-state-university.json` and **Faculty of Agricultural Science** data

### CODE REVIEW

File: public/data.type.ts
Comment (kaf-lamed-beyt) [inline public/data.type.ts:1]: Awesome work! 

files that go into the public folder in a Next.js project are mostly assets &mdash; images, fonts, svgs, icons, etc. 

Let's move this into `types/university.type.ts`

**Review by kaf-lamed-beyt — CHANGES_REQUESTED** [review]
FIne work so far! @stephengade. 

But the validation workflow for the universities failed because you didn't include this new property in the JSON schema

**Comment by stephengade** [comment]
fixed @kaf-lamed-beyt

**Comment by kaf-lamed-beyt** [comment]
Great!

File: public/data.type.ts
Comment (stephengade) [reply public/data.type.ts:1]: done

File: public/types/university.type.ts
Comment (kaf-lamed-beyt) [inline public/types/university.type.ts:1]: @stephengade, i meant for you to move it outside the public folder.

**Review by kaf-lamed-beyt — CHANGES_REQUESTED** [review]
kindly take a look at the comment

**Comment by kaf-lamed-beyt** [comment]
@stephengade i opened a PR to your repo regarding this issue. Please merge when you can.

File: public/types/university.type.ts
Comment (stephengade) [reply public/types/university.type.ts:1]: Thanks for the fix

**Pull request merged**